api/src/wfl/api/routes.clj

```diff
@@ -138,17 +138,25 @@
 (defn exception-handler
   "Top level exception handler. Prefer to use status and message
    from EXCEPTION and fallback to the provided STATUS and MESSAGE."
-  [status _ exception _]
+  [status message exception {:keys [uri] :as _request} return-trace]
   {:status (or (:status (ex-data exception)) status)
-   :body   "An internal error has occurred during this request. The development team has been notified of this error."})
+   :body   (merge {:message message}
+                  (if return-trace
+                    (-> (when-let [cause (.getCause exception)]
+                          {:cause (ExceptionUtils/getRootCauseMessage cause)})
+                        (merge {:uri     uri
+                                :message (or (.getMessage exception) message)
+                                :details (-> exception ex-data (dissoc :status))}))
+                    {}))})
 
 (defn logging-exception-handler
   "Like [[exception-handler]] but also log information about the exception."
-  [status message exception {:keys [uri] :as request}]
-  (let [{:keys [body status] :as result}
-        (exception-handler status message exception request)]
-    (log/error (format "Server %s error at occurred at %s :" status uri))
-    (log/error (str (util/make-map exception body)))
+  [status message labels severity return-trace exception request]
+  (let [{:keys [body] :as result}
+        (exception-handler status message exception request return-trace)]
+    (case severity
+      :warning (log/log :warning (str (util/make-map exception body)) :logging.googleapis.com/labels labels)
+      :error (log/log :error (str (util/make-map exception body)) :logging.googleapis.com/labels labels))
     result))
```

Comment:
I really just put back code that was removed, so this can be updated based on what we think is useful. I was fighting with the severity, as it would just print "SEVERITY" when the argument was passed, but hadn't actually tried binding.

Reply:
Yes. I saw that problem, and believe what you have is OK for now.